src/LoggerFactory.ts

```diff
@@ -0,0 +1,101 @@
+import { Logger } from './logger';
+
+export enum Level {
+  Debug = 1,
+  Info = 2,
+  Warn = 3,
+  Error = 4,
+  Critical = 5
+}
+
+interface Slf {
+  _chain: Middleware[];
+  _queued: Event[];
+  _factory: Factory | null;
+  _logLevel: Level | null;
+  hasWarned: boolean;
+}
+
+const __slf: Slf = {
+  _chain: [],
+  _queued: [],
+  _factory: null,
+  _logLevel: null,
+  hasWarned: false
+};
+
+export interface Event {
+  timeStamp: number;
+  params: any[];
+  name: string;
+  level: string;
+}
+
+export interface Factory {
+  (...events: Event[]): void;
+}
+
+export interface NextFunc {
+  (error: Error | null, event: Event): void;
+}
+
+export interface Middleware {
+  (event: Event, next: NextFunc): void;
+}
+
+export class LoggerFactory {
+  static getLogger(name: string) {
+    let sink;
+    if (__slf._factory) {
+      sink = __slf._factory;
+    } else if (!__slf.hasWarned) {
+      __slf.hasWarned = true;
+      console.log('Warning SLF: No LoggerFactory installed');
+    }
+    if (!sink) {
+      sink = (...args: Event[]) => {
+        if (__slf._factory) {
+          __slf._factory(...args);
+        } else {
+          __slf._queued[__slf._queued.length % 100] = args[0];
```

Comment:
Why is the logic changed? Before, it assigned `args`, no it assigns `args[0]`.

Reply:
If you assign `args` to `_queued` you get a type error, because `_queued` is supposed to be an array of `Event` if you look at how it's used.